torch/onnx/symbolic.py

```diff
@@ -911,9 +941,20 @@ def symbolic(g, input, all_weights, h0, batch_sizes):
                 weight_ih_b, weight_hh_b, bias_ih_b, bias_hh_b = \
                     [reform_weights(g, w, hidden_size, [(1, 2), (0, 1), (2, 3)]) for w in all_weights[2 * i + 1]]
 
-                weight_ih = g.op('Concat', weight_ih_f, weight_ih_b, axis_i=0)
-                weight_hh = g.op('Concat', weight_hh_f, weight_hh_b, axis_i=0)
-                bias_concat = g.op('Concat', bias_ih_f, bias_hh_f, bias_ih_b, bias_hh_b, axis_i=0)
+                weight_ih = g.op('Concat',
+                                 g.op('Unsqueeze', weight_ih_f, axes_i=[0]),
+                                 g.op('Unsqueeze', weight_ih_b, axes_i=[0]),
+                                 axis_i=0)
+                weight_hh = g.op('Concat',
+                                 g.op('Unsqueeze', weight_hh_f, axes_i=[0]),
+                                 g.op('Unsqueeze', weight_hh_b, axes_i=[0]),
+                                 axis_i=0)
+                bias_f = g.op('Concat', bias_ih_f, bias_hh_f, axis_i=0)
+                bias_b = g.op('Concat', bias_ih_b, bias_hh_b, axis_i=0)
+                bias_concat = g.op('Concat',
+                                   g.op('Unsqueeze', bias_f, axes_i=[0]),
+                                   g.op('Unsqueeze', bias_b, axes_i=[0]),
+                                   axis_i=0)
```

Comment:
The diff looks fairly repetitive, is there really no duplication going on here?

Reply:
There is duplication, which I plan to fix but not in this diff. I recently fixed the analogous code duplication on the onnx-caffe2 side here https://github.com/pytorch/pytorch/pull/7263